.github/workflows/pypi-release.yaml

```
@@ -0,0 +1,97 @@
+name: Build and Upload to PyPI
+on:
+  release:
+    types:
+      - published
+  push:
+    tags:
+      - 'v*'
+
```

we could also add the following to catch issues before releases 

```suggestion
    branches:
      - master

  pull_request:
    branches:
      - master

concurrency:
  group: ${{ github.workflow }}-${{ github.ref }}
  cancel-in-progress: true
```